pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -640,13 +641,16 @@ protected BrokerResponse handleRequest(long requestId, String query, @Nullable S
         }
         exceptions.add(QueryException.getException(QueryException.BROKER_SEGMENT_UNAVAILABLE_ERROR, errorMessage));
         _brokerMetrics.addMeteredTableValue(rawTableName, BrokerMeter.BROKER_RESPONSES_WITH_UNAVAILABLE_SEGMENTS, 1);
+        // If segments are unavailable, the result should be considered partial
+        isPartialResult = true;
```

Comment:
We don't need to update it here. We may set the value in the end by checking if the response contains exceptions